Enabling TLS for privacy results in exception

### Steps to Reproduce (Bug)
1. Generate a client certificate and key for Besu
2. Generate PKCS12 keystore and set password
3. Add keystore password to a file
4. Configure the keystore using the --privacy-tls-keystore-file option.
5. Configure the file that contains the password to decrypt the keystore using the --privacy-tls-keystore-password-file option.
6. Configure a fingerprint file - containing fingerprinbt from Orion node configured for client connection TLS --privacy-tls-known-enclave-file
7. Enable TLS for privacy --privacy-tls-enabled=true
8. Start besu

**Expected behavior:** 
Server should start and should be able to 'communicate' over TLS to Orion.

**Actual behavior:** 
Server fails to start with following exception
picocli.CommandLine$ParameterException: java.io.IOException: keystore password was incorrect
	at org.hyperledger.besu.cli.BesuCommand.run(BesuCommand.java:928)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1769)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:2322)
	at org.hyperledger.besu.cli.util.ConfigOptionSearchAndRunHandler.handle(ConfigOptionSearchAndRunHandler.java:61)
	at org.hyperledger.besu.cli.util.ConfigOptionSearchAndRunHandler.handle(ConfigOptionSearchAndRunHandler.java:31)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:2322)
	at org.hyperledger.besu.cli.BesuCommand.parse(BesuCommand.java:1036)
	at org.hyperledger.besu.cli.BesuCommand.parse(BesuCommand.java:911)
	at org.hyperledger.besu.Besu.main(Besu.java:49)
Caused by: picocli.CommandLine$ExecutionException: java.io.IOException: keystore password was incorrect
	at org.hyperledger.besu.cli.BesuCommand.buildController(BesuCommand.java:1205)
	at org.hyperledger.besu.cli.BesuCommand.controller(BesuCommand.java:1197)
	at org.hyperledger.besu.cli.BesuCommand.run(BesuCommand.java:922)
	... 13 more
Caused by: io.vertx.core.VertxException: java.io.IOException: keystore password was incorrect
	at io.vertx.core.net.impl.SSLHelper.createContext(SSLHelper.java:303)
	at io.vertx.core.net.impl.SSLHelper.getContext(SSLHelper.java:478)
	at io.vertx.core.net.impl.SSLHelper.validate(SSLHelper.java:503)
	at io.vertx.core.http.impl.HttpClientImpl.<init>(HttpClientImpl.java:133)
	at io.vertx.core.impl.VertxImpl.createHttpClient(VertxImpl.java:317)
	at org.hyperledger.besu.enclave.EnclaveFactory.createVertxEnclave(EnclaveFactory.java:104)
	at org.hyperledger.besu.ethereum.core.PrivacyParameters$Builder.build(PrivacyParameters.java:259)
	at org.hyperledger.besu.cli.BesuCommand.privacyParameters(BesuCommand.java:1635)
	at org.hyperledger.besu.cli.BesuCommand.getControllerBuilder(BesuCommand.java:1230)
	at org.hyperledger.besu.cli.BesuCommand.buildController(BesuCommand.java:1203)
	... 15 more
Caused by: java.io.IOException: keystore password was incorrect
	at java.base/sun.security.pkcs12.PKCS12KeyStore.engineLoad(PKCS12KeyStore.java:2108)
	at java.base/sun.security.util.KeyStoreDelegator.engineLoad(KeyStoreDelegator.java:222)
	at java.base/java.security.KeyStore.load(KeyStore.java:1479)
	at io.vertx.core.net.impl.KeyStoreHelper.loadJKSOrPKCS12(KeyStoreHelper.java:286)
	at io.vertx.core.net.impl.KeyStoreHelper.create(KeyStoreHelper.java:95)
	at io.vertx.core.net.KeyCertOptions.getKeyManagerFactory(KeyCertOptions.java:53)
	at io.vertx.core.net.impl.SSLHelper.getKeyMgrFactory(SSLHelper.java:308)
	at io.vertx.core.net.impl.SSLHelper.createContext(SSLHelper.java:253)
	... 24 more
Caused by: java.security.UnrecoverableKeyException: failed to decrypt safe contents entry: java.lang.IllegalStateException: password has been cleared
	... 32 more

**Frequency:** 
Every time

### Versions
* Software version: besu/v1.4.1/linux-x86_64/oracle_openjdk-java-11
* Java version: openjdk 11.0.4 2019-07-16 LTS
* OS Name & Version: CentOS Linux release 7.7.1908 (Core)

### Additional Information

Looks like bug in EnclaveFactory.java in package org.hyperledger.besu.enclave 

The function readSecretFromFile is reading the password file as bytes not a line... so you get EOL characters 

Simple fix:
return Files.readAllLines(path, Charsets.UTF_8).get(0);

--- a/enclave/build.gradle
+++ b/enclave/build.gradle
@@ -14,6 +14,7 @@ dependencies {
   testImplementation project(':testutil')
 
   testImplementation 'junit:junit'
+  testImplementation 'org.assertj:assertj-core'
 
   // integration test dependencies.
   integrationTestImplementation project(':testutil')

--- a/enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java
+++ b/enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java
@@ -14,7 +14,6 @@
  */
 package org.hyperledger.besu.enclave;
 
-import com.google.common.io.Files;
 import org.hyperledger.besu.util.InvalidConfigurationException;
 
 import java.io.IOException;
@@ -24,11 +23,11 @@ import java.nio.file.AccessDeniedException;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 
+import com.google.common.io.Files;
 import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpClientOptions;
 import io.vertx.core.net.PfxOptions;
 import org.apache.tuweni.net.tls.VertxTrustOptions;
-
 
 public class EnclaveFactory {
 
@@ -114,7 +113,8 @@ public class EnclaveFactory {
   }
 
   static String readSecretFromFile(final Path path) throws IOException {
-    final String password = Files.asCharSource(path.toFile(), StandardCharsets.UTF_8).readFirstLine();
+    final String password =
+        Files.asCharSource(path.toFile(), StandardCharsets.UTF_8).readFirstLine();
     if (password == null || password.isEmpty()) {
       throw new InvalidConfigurationException("Keystore password file is empty: " + path);
     }

--- a/enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java
+++ b/enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java
@@ -14,20 +14,21 @@
  */
 package org.hyperledger.besu.enclave;
 
+import com.google.common.io.Files;
 import org.hyperledger.besu.util.InvalidConfigurationException;
 
 import java.io.IOException;
 import java.net.URI;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.AccessDeniedException;
-import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 
-import com.google.common.base.Charsets;
 import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpClientOptions;
 import io.vertx.core.net.PfxOptions;
 import org.apache.tuweni.net.tls.VertxTrustOptions;
+
 
 public class EnclaveFactory {
 
@@ -112,8 +113,11 @@ public class EnclaveFactory {
     return new PfxOptions().setPassword(password).setPath(keystoreFile.toString());
   }
 
-  private static String readSecretFromFile(final Path path) throws IOException {
-    final byte[] fileContent = Files.readAllBytes(path);
-    return new String(fileContent, Charsets.UTF_8);
+  static String readSecretFromFile(final Path path) throws IOException {
+    final String password = Files.asCharSource(path.toFile(), StandardCharsets.UTF_8).readFirstLine();
+    if (password == null || password.isEmpty()) {
+      throw new InvalidConfigurationException("Keystore password file is empty: " + path);
+    }
+    return password;
   }
 }

--- a/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
+++ b/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
@@ -1,43 +1,43 @@
 package org.hyperledger.besu.enclave;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
 import org.hyperledger.besu.util.InvalidConfigurationException;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 public class EnclaveFactoryTest {
-    @ClassRule
-    public static final TemporaryFolder temporaryFolder = new TemporaryFolder();
-    private static final String EMPTY_FILE_ERROR_MSG = "Keystore password file is empty: %s";
+  @ClassRule public static final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private static final String EMPTY_FILE_ERROR_MSG = "Keystore password file is empty: %s";
 
-    @Test
-    public void passwordCanBeReadFromFile() throws IOException {
-        final File passwordFile = temporaryFolder.newFile();
-        Files.writeString(passwordFile.toPath(), "test" + System.lineSeparator() + "test2");
-        assertThat(EnclaveFactory.readSecretFromFile(passwordFile.toPath())).isEqualTo("test");
-    }
+  @Test
+  public void passwordCanBeReadFromFile() throws IOException {
+    final File passwordFile = temporaryFolder.newFile();
+    Files.writeString(passwordFile.toPath(), "test" + System.lineSeparator() + "test2");
+    assertThat(EnclaveFactory.readSecretFromFile(passwordFile.toPath())).isEqualTo("test");
+  }
 
-    @Test
-    public void emptyFileThrowsException() throws IOException {
-        final File passwordFile = temporaryFolder.newFile();
-        assertThatExceptionOfType(InvalidConfigurationException.class)
-                .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
-                .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
-    }
+  @Test
+  public void emptyFileThrowsException() throws IOException {
+    final File passwordFile = temporaryFolder.newFile();
+    assertThatExceptionOfType(InvalidConfigurationException.class)
+        .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
+        .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
+  }
 
-    @Test
-    public void fileWithOnlyEoLThrowsException() throws IOException {
-        final File passwordFile = temporaryFolder.newFile();
-        Files.writeString(passwordFile.toPath(), System.lineSeparator());
-        assertThatExceptionOfType(InvalidConfigurationException.class)
-                .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
-                .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
-    }
+  @Test
+  public void fileWithOnlyEoLThrowsException() throws IOException {
+    final File passwordFile = temporaryFolder.newFile();
+    Files.writeString(passwordFile.toPath(), System.lineSeparator());
+    assertThatExceptionOfType(InvalidConfigurationException.class)
+        .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
+        .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
+  }
 }

--- a/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
+++ b/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
@@ -1,3 +1,17 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
 package org.hyperledger.besu.enclave;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
+++ b/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java
@@ -1,0 +1,43 @@
+package org.hyperledger.besu.enclave;
+
+import org.hyperledger.besu.util.InvalidConfigurationException;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class EnclaveFactoryTest {
+    @ClassRule
+    public static final TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private static final String EMPTY_FILE_ERROR_MSG = "Keystore password file is empty: %s";
+
+    @Test
+    public void passwordCanBeReadFromFile() throws IOException {
+        final File passwordFile = temporaryFolder.newFile();
+        Files.writeString(passwordFile.toPath(), "test" + System.lineSeparator() + "test2");
+        assertThat(EnclaveFactory.readSecretFromFile(passwordFile.toPath())).isEqualTo("test");
+    }
+
+    @Test
+    public void emptyFileThrowsException() throws IOException {
+        final File passwordFile = temporaryFolder.newFile();
+        assertThatExceptionOfType(InvalidConfigurationException.class)
+                .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
+                .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
+    }
+
+    @Test
+    public void fileWithOnlyEoLThrowsException() throws IOException {
+        final File passwordFile = temporaryFolder.newFile();
+        Files.writeString(passwordFile.toPath(), System.lineSeparator());
+        assertThatExceptionOfType(InvalidConfigurationException.class)
+                .isThrownBy(() -> EnclaveFactory.readSecretFromFile(passwordFile.toPath()))
+                .withMessage(EMPTY_FILE_ERROR_MSG, passwordFile);
+    }
+}

